### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by Donard97**
**I created this PR to check if our project was as requested.**

_This is a To-dos app that was created with React. In this milestone, we had to write unit tests for the app. 📝_

### Project Requirement:

> - Used Jest and React Testing Library   ✔️ 
> - Wrote unit test for files operate.js and calculate.js ✔️ 
> -  Used Jest snapshots to test the components ✔️ 
> - Used React Testing Library to simulate user interaction ✔️ 

<hr>

_Feel free to review my project and let me know if there are any essential changes 💪 ._

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "react-dom": "^17.0.2",
     "react-router-dom": "^5.3.0",
     "react-scripts": "4.0.3",
+    "react-test-renderer": "^17.0.2",
     "web-vitals": "^1.0.1"
   },
   "scripts": {
```
Comment (EricMbouwe) [inline package.json:20]: - [OPTIONAL] Using this option `"test": "react-scripts test --verbose",` helps seeing your tests description.

File: src/components/Calculator.test.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Calculator from './calculator';
+
+it('Calculator component is working', () => {
+  const tree = renderer.create(<Calculator />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (EricMbouwe) [inline src/components/Calculator.test.js:8]: - Excellent!

File: src/components/Navbar.test.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+it('Navbar component is working', () => {
+  const tree = renderer
+    .create(
+      <Router>
+        <Navbar />
+      </Router>,
+    )
+    .toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (EricMbouwe) [inline src/components/Navbar.test.js:15]: - Good!

File: src/logic/calculate.test.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import calculate from './calculate';
+
+describe('calculate function', () => {
+  test('returns undefined for calculation with only zero', () => {
```
Comment (EricMbouwe) [inline src/logic/calculate.test.js:4]: - Good job done here!

File: src/logic/operate.test.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import operate from './operate';
+
+describe('Operate Function', () => {
+  test('correctly returns 5 when 3 + 2 is summed', () => {
```
Comment (EricMbouwe) [inline src/logic/operate.test.js:4]: - Excellent!

**Review by EricMbouwe — APPROVED** [review]
Hi @Donard97, 

# Status: Approved✔️
Excellent job on this project. Keep it up 👏
Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json (hunk 1 above)
Comment (Donard97) [reply package.json:20]: Thank you for this suggestion. I will do it.

File: src/logic/calculate.test.js (hunk 4 above)
Comment (Donard97) [reply src/logic/calculate.test.js:4]: Thank you!

File: src/logic/operate.test.js (hunk 5 above)
Comment (Donard97) [reply src/logic/operate.test.js:4]: Thank you!

File: src/components/Calculator.test.js (hunk 2 above)
Comment (Donard97) [reply src/components/Calculator.test.js:8]: Thank you!

**Pull request merged**